src/core/ddsi/src/q_entity.c

```diff
@@ -2324,8 +2470,24 @@ static void connect_writer_with_proxy_reader (struct writer *wr, struct proxy_re
     writer_qos_mismatch (wr, reason);
     return;
   }
-  proxy_reader_add_connection (prd, wr);
-  writer_add_connection (wr, prd);
+  if (q_omg_security_check_remote_reader_permissions (prd, wr->e.gv->config.domainId, wr->c.pp))
```

Comment:
For next time: inverting and reordering the tests gives a much more compact if/else if/else 😄  pattern.